packages/vkui/src/components/ImageBase/ImageBasePositionedComponent/ImageBasePositionedComponent.tsx

```diff
@@ -0,0 +1,191 @@
+  insetBlockEnd?: React.CSSProperties['insetBlockEnd'];
+};
+
+export type PositionedComponentIndentation =
+  | '2xs'
+  | 'xs'
+  | 's'
+  | 'm'
+  | 'l'
+  | 'xl'
+  | '2xl'
+  | '3xl'
+  | '4xl'
+  | number
+  | string;
+
+const positionPlacementClassNames = {
+  'top-start': styles['PositionedComponent--position-topStart'],
+  'top': styles['PositionedComponent--position-top'],
+  'top-end': styles['PositionedComponent--position-topEnd'],
+  'bottom-start': styles['PositionedComponent--position-bottomStart'],
+  'bottom': styles['PositionedComponent--position-bottom'],
+  'bottom-end': styles['PositionedComponent--position-bottomEnd'],
+  'middle-start': styles['PositionedComponent--position-middleStart'],
+  'middle': styles['PositionedComponent--position-middle'],
+  'middle-end': styles['PositionedComponent--position-middleEnd'],
+};
+
+const horizontalIndentClassNames = {
+  '2xs': styles['PositionedComponent--horizontalIndent-2xs'],
+  'xs': styles['PositionedComponent--horizontalIndent-xs'],
+  's': styles['PositionedComponent--horizontalIndent-s'],
+  'm': styles['PositionedComponent--horizontalIndent-m'],
+  'l': styles['PositionedComponent--horizontalIndent-l'],
+  'xl': styles['PositionedComponent--horizontalIndent-xl'],
+  '2xl': styles['PositionedComponent--horizontalIndent-2xl'],
+  '3xl': styles['PositionedComponent--horizontalIndent-3xl'],
+  '4xl': styles['PositionedComponent--horizontalIndent-4xl'],
+};
+
+const verticalIndentClassNames = {
+  '2xs': styles['PositionedComponent--verticalIndent-2xs'],
+  'xs': styles['PositionedComponent--verticalIndent-xs'],
+  's': styles['PositionedComponent--verticalIndent-s'],
+  'm': styles['PositionedComponent--verticalIndent-m'],
+  'l': styles['PositionedComponent--verticalIndent-l'],
+  'xl': styles['PositionedComponent--verticalIndent-xl'],
+  '2xl': styles['PositionedComponent--verticalIndent-2xl'],
+  '3xl': styles['PositionedComponent--verticalIndent-3xl'],
+  '4xl': styles['PositionedComponent--verticalIndent-4xl'],
+};
+
+export interface ImageBasePositionedComponentProps
+  extends HTMLAttributesWithRootRef<HTMLDivElement> {
+  /**
+   * Позиция компонента относительно родителя
+   */
+  position: PositionedComponentPlacement | PositionedComponentPosition;
+  /**
+   * Отступ компонента от края контейнера по горизонтали
+   */
+  horizontalIndent?: PositionedComponentIndentation;
+  /**
+   * Отступ компонента от края контейнера по вертикали
+   */
+  verticalIndent?: PositionedComponentIndentation;
+  /**
+   * Режим отображения компонента:
+   *
+   * - `"always"`: Всегда
+   * - `"on-image-hover"`: При наведении на картинку
+   */
+  visibility?: 'always' | 'on-image-hover';
+  /**
+   * `ref` на контейнер, в который встраивается компонент.
+   * Используется в случае использования компонента отдельно от `Image`
+   */
+  containerRef?: React.RefObject<HTMLElement | null>;
+}
+
+export const ImageBasePositionedComponent = ({
```

Comment:
```suggestion
export const ImageBaseFloatElement = ({
```
в голову приходит вот такое название

Reply:
Переименовал